plotjuggler_app/nlohmann_parsers.h

```diff
@@ -113,21 +114,43 @@ class NlohmannParserCreator : public ParserFactoryPlugin
   NlohmannParserCreator()
   {
     _checkbox_use_timestamp = new QCheckBoxClose("use field as timestamp if available");
+    loadSettings();
   }
 
   template <typename ParserT>
   MessageParserPtr createParserImpl(const std::string& topic_name, PlotDataMapRef& data)
   {
+    saveSettings();
+
     std::string timestamp_name = _checkbox_use_timestamp->lineedit->text().toStdString();
     return std::make_shared<ParserT>(
         topic_name, data, _checkbox_use_timestamp->isChecked(), timestamp_name);
   }
 
   virtual QWidget* optionsWidget()
   {
+    loadSettings();
+
     return _checkbox_use_timestamp;
   }
 
+  virtual void loadSettings()
+  {
+    QSettings settings;
+    bool checked = settings.value("NlohmannParser.timestampEnabled", false).toBool();
+    QString field = settings.value("NlohmannParser.timestampFieldName", "").toString();
+
+    _checkbox_use_timestamp->setChecked(checked);
+    _checkbox_use_timestamp->lineedit->setText(field);
+  }
+
+  virtual void saveSettings()
+  {
+    QSettings settings;
+    settings.setValue("NlohmannParser.timestampEnabled", _checkbox_use_timestamp->isChecked());
+    settings.setValue("NlohmannParser.timestampFieldName", _checkbox_use_timestamp->lineedit->text());
+  }
+
```

Comment:
```suggestion
  virtual void loadSettings()
  {
    QSettings settings;
    QString prefix = QString("NlohmannParser.") + QString(encoding());
    bool checked = settings.value(prefix + ".timestampEnabled", false).toBool();
    QString field = settings.value(prefix + ".timestampFieldName", "").toString();

    _checkbox_use_timestamp->setChecked(checked);
    _checkbox_use_timestamp->lineedit->setText(field);
  }

  virtual void saveSettings()
  {
    QSettings settings;
    QString prefix = QString("NlohmannParser.") + QString(encoding());
    settings.setValue(prefix + ".timestampEnabled", _checkbox_use_timestamp->isChecked());
    settings.setValue(prefix + ".timestampFieldName", _checkbox_use_timestamp->lineedit->text());
  }

```

Reply:
shall we have different parameters, based on encoding?

Reply:
Sure, makes sense! That was also a neat way of doing it without duplication.

Reply:
Hmm, that code change actually crashes on my computer. I'm assuming `encoding()` of the derived class is not callable from the base class constructor.

Perhaps `loadSettings()` could be removed from the constructor, which I mostly added for "good measure", in case the options were to be used within the factory class in the future.

Reply:
I solved that by moving the `encoding()` function up to `NlohmannParserCreator`. Hope that is OK!